dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+        );
+      if (numPaths != frame.paths.length)
+        throw new Exception(
+            'All animation frames must have the same number of paths, '
+                'first frame has $numPaths paths'
+                'frame $i has ${frame.paths.length} paths'
+        );
+    }
+  }
+
+  String toDart(String className, String varName) {
+    String result = 'const $className $varName = const $className(\n';
+    result += '${kIndent}const Size(${size.x}, ${size.y}),\n';
+    result += '${kIndent}const <_PathFrames> [\n';
+    for (PathAnimation path in paths)
+      result += path.toDart();
+    result += '$kIndent],\n';
+    result += ');';
+    return result;
+  }
+}
+
+/// Represents the animation of a single path.
+class PathAnimation {
+  const PathAnimation(this.commands, {@required this.opacities});
+
+  factory PathAnimation.fromFrameData(List<FrameData> frames, int pathIdx) {
+    if (frames.isEmpty)
+      return const PathAnimation(const <PathCommandAnimation> [], opacities: const <double> []);
+
+    final List<PathCommandAnimation> commands = <PathCommandAnimation>[];
+    for (int commandIdx = 0; commandIdx < frames[0].paths[pathIdx].commands.length; commandIdx++) {
+      final int numPointsInCommand = frames[0].paths[pathIdx].commands[commandIdx].points.length;
+      final List<List<Point<double>>> points = new List<List<Point<double>>>(numPointsInCommand);
+      for (int j = 0; j < numPointsInCommand; j++)
+        points[j] = <Point<double>>[];
+      final String commandType = frames[0].paths[pathIdx].commands[commandIdx].type;
+      for (int i = 0; i < frames.length; i++) {
+        final FrameData frame = frames[i];
+        final String currentCommandType = frame.paths[pathIdx].commands[commandIdx].type;
+        if (commandType != currentCommandType)
+          throw new Exception(
+              'Paths must be built from the same commands in all frames'
+              'command $commandIdx at frame 0 was of type \'$commandType\''
+              'command $commandIdx at frame $i was of type \'$currentCommandType\''
+          );
+        for (int j = 0; j < numPointsInCommand; j++)
+          points[j].add(frame.paths[pathIdx].commands[commandIdx].points[j]);
+      }
+      commands.add(new PathCommandAnimation(commandType, points));
+    }
+
+    final List<double> opacities =
+      frames.map<double>((FrameData d) => d.paths[pathIdx].opacity).toList();
+
+    return new PathAnimation(commands, opacities: opacities);
+  }
+
+  /// List of commands for drawing the path.
+  final List<PathCommandAnimation> commands;
+  /// The path opacity for each animation frame.
+  final List<double> opacities;
+
+  @override
+  String toString() {
+    return 'PathAnimation(commands: $commands, opacities: $opacities)';
+  }
+
+  String toDart() {
+    String result = '${kIndent*2}const _PathFrames(\n';
+    result += '${kIndent*3}opacities: const <double> [\n';
```

Comment:
nit: extraneous space before `[` in output

Reply:
done